client/Makefile.am

```diff
@@ -86,6 +86,8 @@ ipa_join_LDADD = 		\
 	$(LDAP_LIBS)		\
 	$(SASL_LIBS)		\
 	$(XMLRPC_LIBS)		\
+	$(JANSSON_LIBS)		\
+	$(LIBCURL_LIBS)		\
```

Comment:
You have defined automake condition for choosing which join method to use but the condition is not used in the makefile that specifies dependencies. It means both sets of libraries will be pulled in. We need to choose one.
I think this part needs to be split into a separate condition and assign either set of libraries to a variable that then gets referenced here.

Reply:
I was also under the impression that this would be necessary. However when I look at the linker flags for both cases, only the necessary flags are used.

JSON:
```
[...] ../util/.libs/libutil.a [...] -ljansson -lcurl -lpopt
```

XML:
```
[...] ../util/.libs/libutil.a [...] -lxmlrpc -lxmlrpc_client -lxmlrpc_util -lpopt
```

Reply:
When JSON is used then XMLRPC_CFLAGS and XLRPC_LIBS are both empty:
```
XMLRPC_CFLAGS =
XMLRPC_LIBS =
```
And the inverse is true when --with-ipa-join-xml is selected. the JSAN values are empty.

This looks ok to me. It might be slightly confusing looking only at the client Makefile but in the source-level config.log it's pretty clear.